server/odc-service/src/main/java/com/oceanbase/odc/service/connection/logicaldatabase/LogicalDatabaseService.java

```diff
@@ -0,0 +1,228 @@
+    private DBObjectRepository dbObjectRepository;
+
+    @Autowired
+    private TableMappingRepository tableMappingRepository;
+
+    @Autowired
+    private ConnectionConfigRepository connectionRepository;
+
+    @Autowired
+    private AuthenticationFacade authenticationFacade;
+
+    @Autowired
+    private EnvironmentService environmentService;
+
+    @Autowired
+    private DatabaseService databaseService;
+
+    @Autowired
+    private LogicalTableService tableService;
+
+    @Autowired
+    private LogicalDatabaseSyncManager syncManager;
+
+    @Autowired
+    private DBResourcePermissionHelper permissionHelper;
+
+
+    @Transactional(rollbackFor = Exception.class)
+    public Boolean create(CreateLogicalDatabaseReq req) {
+        preCheck(req);
+
+        Long organizationId = authenticationFacade.currentOrganizationId();
+        DatabaseEntity basePhysicalDatabase = databaseRepository
+                .findById(req.getPhysicalDatabaseIds().iterator().next()).orElseThrow(() -> new NotFoundException(
+                        ResourceType.ODC_DATABASE, "id", req.getPhysicalDatabaseIds().iterator().next()));
+        ConnectionEntity baseConnection = connectionRepository.findById(basePhysicalDatabase.getConnectionId())
+                .orElseThrow(() -> new NotFoundException(
+                        ResourceType.ODC_CONNECTION, "id", basePhysicalDatabase.getConnectionId()));
+        DatabaseEntity logicalDatabase = new DatabaseEntity();
+        logicalDatabase.setProjectId(req.getProjectId());
+        logicalDatabase.setName(req.getName());
+        logicalDatabase.setAlias(req.getAlias());
+        logicalDatabase.setEnvironmentId(basePhysicalDatabase.getEnvironmentId());
+        logicalDatabase.setDatabaseId(StringUtils.uuid());
+        logicalDatabase.setType(DatabaseType.LOGICAL);
+        logicalDatabase.setDialectType(baseConnection.getDialectType());
+        logicalDatabase.setSyncStatus(DatabaseSyncStatus.INITIALIZED);
+        logicalDatabase.setObjectSyncStatus(DBObjectSyncStatus.INITIALIZED);
+        logicalDatabase.setExisted(true);
+        logicalDatabase.setOrganizationId(organizationId);
+        DatabaseEntity savedLogicalDatabase = databaseRepository.saveAndFlush(logicalDatabase);
+
+        List<DatabaseMappingEntity> mappings = new ArrayList<>();
+        req.getPhysicalDatabaseIds().stream().forEach(physicalDatabaseId -> {
+            DatabaseMappingEntity relation = new DatabaseMappingEntity();
+            relation.setLogicalDatabaseId(savedLogicalDatabase.getId());
+            relation.setPhysicalDatabaseId(physicalDatabaseId);
+            relation.setOrganizationId(organizationId);
+            mappings.add(relation);
+        });
+        databaseMappingRepository.batchCreate(mappings);
+
+        return true;
+    }
+
+    public DetailLogicalDatabaseResp detail(@NotNull Long id) {
+        permissionHelper.checkDBPermissions(Collections.singleton(id), DatabasePermissionType.all());
```

Comment:
Why `DatabasePermissionType.all()`? Warning that `DBResourcePermissionHelper.checkDBPermissions` means need both all permission rather than any one of them.
Same as below.

Reply:
I deleted it. We should check the project role rather than database permission here.